tests/unit/test_check_operators/test_string_comparison.py

```diff
@@ -703,3 +703,153 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
         {"target": "target", "comparator": comparator}
     )
     assert result.equals(df.convert_to_series(expected_result))
+
+
+@pytest.mark.parametrize(
+    "data,separator,dataset_type,expected_result",
+    [
```

Comment:
Since these first 4 test cases all have the same separator and dataset_type, why not combine into a single test case, combining the data, and expected result arrays?  Same for combining the last 2 cases into a single test case.